src/Views/FirmwareView.vala

```diff
@@ -0,0 +1,117 @@
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*
+* Authored by: Marius Meisenzahl <mariusmeisenzahl@gmail.com>
+*/
+
+public class About.FirmwareView : Granite.SettingsPage {
+    private Gtk.Stack stack;
+    private Gtk.Grid grid;
+    private Gtk.Grid progress_view;
+    private Gtk.ListBox update_list;
+
+    public FirmwareView () {
+        Object (
+            icon_name: "application-x-firmware",
+            title: _("Firmware")
+        );
+    }
+
+    construct {
+        var progress_alert_view = new Granite.Widgets.AlertView (
+            _("Device is being updated"),
+            _("Do not unplug the device during the update."),
+            "emblem-synchronized"
+        );
+        progress_alert_view.get_style_context ().remove_class (Gtk.STYLE_CLASS_VIEW);
+
+        progress_view = new Gtk.Grid () {
+            margin = 24
+        };
+        progress_view.attach (progress_alert_view, 0, 0, 1, 1);
+
+        var header_icon = new Gtk.Image.from_icon_name ("application-x-firmware", Gtk.IconSize.DIALOG) {
+            pixel_size = 48,
+            valign = Gtk.Align.START
+        };
+
+        var title_label = new Gtk.Label (_("Firmware")) {
+            ellipsize = Pango.EllipsizeMode.END,
+            hexpand = true,
+            selectable = true,
+            xalign = 0
+        };
+        title_label.get_style_context ().add_class (Granite.STYLE_CLASS_H2_LABEL);
+
+        grid = new Gtk.Grid () {
+            column_spacing = 12,
+            row_spacing = 12,
+            margin = 12
+        };
+
+        update_list = new Gtk.ListBox () {
+            vexpand = true,
+            selection_mode = Gtk.SelectionMode.SINGLE
+        };
+
+        var scrolled_window = new Gtk.ScrolledWindow (null, null);
+        scrolled_window.add (update_list);
+
+        var frame = new Gtk.Frame (null);
+        frame.add (scrolled_window);
+
+        grid.attach (header_icon, 0, 0);
+        grid.attach (title_label, 1, 0);
+        grid.attach (frame, 0, 1, 3, 1);
+
+        stack = new Gtk.Stack ();
+        stack.transition_type = Gtk.StackTransitionType.SLIDE_LEFT_RIGHT;
+
+        stack.add (grid);
+        stack.add (progress_view);
+
+        add (stack);
+
+        update_list_view.begin ();
+    }
+
+    private async void update_list_view () {
+        foreach (Gtk.Widget element in update_list.get_children ()) {
```

Comment:
This should be an unowned reference:
```suggestion
        foreach (unowned Gtk.Widget element in update_list.get_children ()) {
```

It would probably also be better to use `widget` as a variable name instead of `element` since this is a `Gtk.Widget`